File: src/jvm/package/src/doc-examples/java/questpdf/docexamples/EnsureSpaceExamples.java
package questpdf.docexamples;

import com.questpdf.fluent.Document;
import com.questpdf.helpers.Colors;
import com.questpdf.helpers.PageSizes;
import com.questpdf.helpers.Placeholders;
import org.junit.jupiter.api.Test;

public class EnsureSpaceExamples extends DocExample {

    @Test
    public void enabledExample() {
        Document
            .create(document -> {
                document.page(page -> {
                    page.size(PageSizes.getA5());
                    page.defaultTextStyle(style -> style.fontSize(20f));
                    page.margin(30f);

                    page.content()
                        .column(column -> {
                            column.item().height(400f).background(Colors.Grey.getLighten3());
                            column.item().height(30f);

                            column.item()
                                .ensureSpace(100f)
                                .table(table -> {
                                    table.columnsDefinition(columns -> {
                                        columns.constantColumn(40f);
                                        columns.relativeColumn();
                                    });

                                    for (var i = 1; i <= 12; i++) {
                                        table.cell().text(i + ".");
                                        table.cell().showEntire().text(Placeholders.sentence());
                                    }
                                });
                        });
                });
            })
            .generatePdf(output("ensure-space-enabled.pdf"));
    }

    @Test
    public void disabledExample() {
        Document
            .create(document -> {
                document.page(page -> {
                    page.size(PageSizes.getA5());
                    page.defaultTextStyle(style -> style.fontSize(20f));
                    page.margin(30f);

                    page.content()
                        .column(column -> {
                            column.item().height(400f).background(Colors.Grey.getLighten3());
                            column.item().height(30f);

                            column.item()
                                .table(table -> {
                                    table.columnsDefinition(columns -> {
                                        columns.constantColumn(40f);
                                        columns.relativeColumn();
                                    });

                                    for (var i = 1; i <= 12; i++) {
                                        table.cell().text(i + ".");
                                        table.cell().text(Placeholders.sentence());
                                    }
                                });
                        });
                });
            })
            .generatePdf(output("ensure-space-disabled.pdf"));
    }
}

File: src/jvm/package/src/doc-examples/kotlin/questpdf/docexamples/EnsureSpaceExamples.kt
package questpdf.docexamples

import com.questpdf.fluent.Document
import com.questpdf.helpers.Colors
import com.questpdf.helpers.PageSizes
import com.questpdf.helpers.Placeholders
import org.junit.jupiter.api.Test

class EnsureSpaceExamples : DocExample() {

    @Test
    fun enabledExample() {
        Document
            .create {
                page {
                    size(PageSizes.A5)
                    defaultTextStyle { fontSize(20f) }
                    margin(30f)

                    content()
                        .column {
                            item().height(400f).background(Colors.Grey.Lighten3)
                            item().height(30f)

                            item()
                                .ensureSpace(100f)
                                .table {
                                    columnsDefinition {
                                        constantColumn(40f)
                                        relativeColumn()
                                    }

                                    for (i in 1..12) {
                                        cell().text("$i.")
                                        cell().showEntire().text(Placeholders.sentence())
                                    }
                                }
                        }
                }
            }
            .generatePdf(output("ensure-space-enabled.pdf"))
    }

    @Test
    fun disabledExample() {
        Document
            .create {
                page {
                    size(PageSizes.A5)
                    defaultTextStyle { fontSize(20f) }
                    margin(30f)

                    content()
                        .column {
                            item().height(400f).background(Colors.Grey.Lighten3)
                            item().height(30f)

                            item()
                                .table {
                                    columnsDefinition {
                                        constantColumn(40f)
                                        relativeColumn()
                                    }

                                    for (i in 1..12) {
                                        cell().text("$i.")
                                        cell().text(Placeholders.sentence())
                                    }
                                }
                        }
                }
            }
            .generatePdf(output("ensure-space-disabled.pdf"))
    }
}
